boto3/resources/factory.py

```diff
@@ -133,9 +133,10 @@ def load_from_definition(self, resource_name,
 
         base_classes = [ServiceResource]
         if self._emitter is not None:
-            self._emitter.emit('creating-resource-class.%s' % cls_name,
-                               class_attributes=attrs,
-                               base_classes=base_classes)
+            self._emitter.emit(
+                'creating-resource-class.%s' % cls_name,
+                class_attributes=attrs, base_classes=base_classes,
+                service_context=service_context, resource_name=resource_name)
```

Comment:
We could potentially pull it off the event name. I didn't want to add it to `CustomModeledAction` since you would essentially be providing that info twice because it is part of the event name.